src/constraints/capacity.jl

```diff
@@ -296,12 +299,15 @@ function add_capacity_constraints!(
     ]
 
     # - Lower limit for flows associated with assets
-    assets_with_non_negative_outgoing_flows = Ap ∪ Acv ∪ As
-    assets_with_non_negative_incoming_flows = Acv ∪ As
-    for row in eachrow(df_flows)
-        if row.from in assets_with_non_negative_outgoing_flows ||
-           row.to in assets_with_non_negative_incoming_flows
-            JuMP.set_lower_bound(flow[row.index], 0.0)
-        end
+    assets_with_non_negative_outgoing_flows_indices =
+        filter(:from => in(Ap ∪ Acv ∪ As), flows_indices; view = true)
+    for row in eachrow(assets_with_non_negative_outgoing_flows_indices)
+        JuMP.set_lower_bound(flow[row.index], 0.0)
+    end
+
+    assets_with_non_negative_incoming_flows_indices =
+        filter(:to => in(Acv ∪ As), flows_indices; view = true)
+    for row in eachrow(assets_with_non_negative_incoming_flows_indices)
+        JuMP.set_lower_bound(flow[row.index], 0.0)
```

Comment:
You should be able to write this with a single filter (or DataFrames.subset, I never know which is best)

```julia
df_with_good_name = DataFrames.subset(
  flows_indices,
  [:from, :to] => ByRow((from, to) -> from in Ap || from in Acv || from in As || to in Acv || to in As),
)
```

I've separated the union into different `in` checks, because the `union` creates a temporary set (AFAIK).